exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/DrillConnectionImpl.java

```diff
@@ -362,12 +364,33 @@ public PreparedStatement prepareStatement(String sql, int resultSetType,
                                             int resultSetHoldability) throws SQLException {
     throwIfClosed();
     try {
-      DrillPrepareResult prepareResult = new DrillPrepareResult(sql);
+      DrillRpcFuture<CreatePreparedStatementResp> respFuture = client.createPreparedStatement(sql);
+
+      CreatePreparedStatementResp resp;
+      try {
+        resp = respFuture.get();
+      } catch (InterruptedException e) {
+        // Preserve evidence that the interruption occurred so that code higher up
+        // on the call stack can learn of the interruption and respond to it if it
+        // wants to.
+        Thread.currentThread().interrupt();
+
+        throw new SQLException( "Interrupted", e );
```

Comment:
Since the client connection is still alive, the query would continue to execute on the server and presumably still send back results. Where are the results consumed if an interrupt occurs here?